client/client.go

```diff
@@ -1291,16 +1323,19 @@ func (r *repository) pubKeysToCerts(role data.RoleName, pubKeyList data.KeyList)
 	}
 
 	for i, pubKey := range pubKeyList {
-		privKey, loadedRole, err := r.GetCryptoService().GetPrivateKey(pubKey.ID())
-		if err != nil {
-			return nil, err
-		}
-		if loadedRole != role {
-			return nil, fmt.Errorf("attempted to load root key but given %s key instead", loadedRole)
-		}
-		pubKey, err = rootCertKey(r.gun, privKey)
-		if err != nil {
-			return nil, err
+		// convert to x509 if pubKey is not already one
+		if !utils.IsX509Key(pubKey) {
+			privKey, loadedRole, err := r.GetCryptoService().GetPrivateKey(pubKey.ID())
+			if err != nil {
+				return nil, fmt.Errorf("error converting public key with id %v to private key: %v", pubKey.ID(), err)
```

Comment:
let's change this message: `fmt.Errorf("error retrieving private key for corresponding public key with ID %s: %v", pubKey.ID(), err)`